compiler/src/iree/compiler/PluginAPI/Client.h

```diff
@@ -39,6 +40,11 @@ class PipelineExtensions {
  public:
   virtual ~PipelineExtensions();
 
+  // Add passes to the input preprocessing pipeline, which allows to process the
+  // raw input to IREE.
+  virtual void extendInputConversionPreprocessingPassPipeline(
+      OpPassManager &passManager, InputDialectOptions::Type inputType) {}
```

Comment:
Note: I may remove InputDialectOptions::Type and instead have it be more fine grained information (e.g., handle input with both StableHLO & TMTensor, rather than exclusive) and the pipelines would all run after the auto analysis. This would look like the struct in auto conversion input pipeline currently. In your case you are just using this as an optimization to not invoke the conversion pipeline I believe - although would be good to have verification that that preprocessing is a NOP if the input is (say) TOSA input.